internal/tiltfile/extension/store.go

```diff
@@ -0,0 +1,71 @@
+package extension
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/windmilleng/tilt/internal/tiltfile"
+)
+
+const extensionDirName = "tilt_modules"
+
+type Store interface {
+	// Stat is used to check if an extension exists before fetching it
+	Stat(ctx context.Context, moduleName string) (string, error)
+	Write(ctx context.Context, contents ModuleContents) (string, error)
+}
+
+// TODO(dmiller): should this include an integrity hash?
+type ModuleContents struct {
+	Name             string
+	TiltfileContents string
+	GitCommitHash    string
+	Source           string
```

Comment:
ha, when I read "Source", I interpreted it as the source code of the extension, and was trying to figure out how it differed from `TiltfileContents`

maybe we want a `type ExtensionRegistry string`?